cargo-spatial/src/download.rs

```diff
@@ -0,0 +1,256 @@
+// Conditionally re-export the correct version of download_cli dependent on platform.
+#[cfg(target_os = "linux")]
+pub use self::linux::*;
+#[cfg(any(target_os = "windows", target_os = "macos"))]
+pub use self::win_osx::*;
+
+use crate::config::Config;
+use crate::opt::DownloadSdk;
+use log::*;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process,
+};
+
+enum SpatialWorkerSdkPackage {
+    CApiWin,
+    CApiMac,
+    CApiLinux,
+}
+
+impl SpatialWorkerSdkPackage {
+    fn get_package_name(&self) -> &str {
+        match self {
+            SpatialWorkerSdkPackage::CApiWin => "c-static-x86_64-msvc_mt-win32",
+            SpatialWorkerSdkPackage::CApiMac => "c-static-x86_64-clang_libcpp-macos",
+            SpatialWorkerSdkPackage::CApiLinux => "c-static-x86_64-gcc_libstdcpp_pic-linux",
+        }
+    }
+
+    fn get_relative_target_directory(&self) -> &str {
+        match self {
+            SpatialWorkerSdkPackage::CApiWin => "win",
+            SpatialWorkerSdkPackage::CApiMac => "macos",
+            SpatialWorkerSdkPackage::CApiLinux => "linux",
+        }
+    }
+}
+
+enum SpatialToolsPackage {
```

Comment:
Would be good to derive some common traits here, e.g. `derive(Debug, Clone, Copy, PartialEq, Eq, Hash)`. Same goes for all the other enums you define here.